x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/analytics_logic.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+import queryString from 'query-string';
+
+import { KibanaLogic } from '../../../shared/kibana';
+import { HttpLogic } from '../../../shared/http';
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { EngineLogic } from '../engine';
+
+import { AnalyticsData, QueryDetails } from './types';
+
+export interface AnalyticsValues extends AnalyticsData, QueryDetails {
+  dataLoading: boolean;
+}
+
+export interface AnalyticsActions {
+  onAnalyticsUnavailable(): void;
+  onAnalyticsDataLoad(data: AnalyticsData): AnalyticsData;
+  onQueryDataLoad(data: QueryDetails): QueryDetails;
+  loadAnalyticsData(): void;
+  loadQueryData(query: string): string;
+}
+
+export const AnalyticsLogic = kea<MakeLogicType<AnalyticsValues, AnalyticsActions>>({
+  path: ['enterprise_search', 'app_search', 'analytics_logic'],
+  actions: () => ({
+    onAnalyticsUnavailable: true,
+    onAnalyticsDataLoad: (data) => data,
+    onQueryDataLoad: (data) => data,
+    loadAnalyticsData: true,
+    loadQueryData: (query) => query,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        loadAnalyticsData: () => true,
+        loadQueryData: () => true,
+        onAnalyticsUnavailable: () => false,
+        onAnalyticsDataLoad: () => false,
+        onQueryDataLoad: () => false,
+      },
+    ],
+    analyticsUnavailable: [
+      false,
+      {
+        onAnalyticsUnavailable: () => true,
+        onAnalyticsDataLoad: (_, { analyticsUnavailable }) => analyticsUnavailable,
+        onQueryDataLoad: (_, { analyticsUnavailable }) => analyticsUnavailable,
+      },
+    ],
```

Comment:
There's a ton more reducers still to import, these are the primarily http/fetch-related ones that I know I'll be using soon.

I'm migrating reducers as-needed to try and catch ones we don't need and can potentially clean up from ent-search.

Reply:
Good call. I like to migrate them as-needed as well.